homeassistant/helpers/entity.py

```diff
@@ -628,6 +634,27 @@ def _name_translation_key(self) -> str | None:
             f".{self.translation_key}.name"
         )
 
+    def _substitue_name_placeholders(self, name: str) -> str:
+        """Substitute placeholders in entity name."""
+        try:
+            return name.format(**self.translation_placeholders)
+        except KeyError:
+            if not self._name_translation_placeholders_reported:
```

Comment:
Didn't we discuss that we should blow up on dev?

Reply:
Yes, we did talk about that. Do you still think that's valid? If yes, I'll make it blow up on the non stable channel.

Reply:
Fixed.